src/compressed_tensors/transforms/utils.py

```diff
@@ -15,7 +15,27 @@
 import torch
 
 
-__all__ = ["apply_matrix_transform"]
+__all__ = ["apply_matrix_transform", "SingletonMatrixRegistry"]
+
+
+class SingletonMatrixRegistry:
```

Comment:
so that all matrices live in a single global key-value store, right?

Reply:
yeah, we can expand this but it seems like there will be a lot of repetition across decoder layers for example
I think if this goes too big in scope, we may have to consider other data stores to handle it